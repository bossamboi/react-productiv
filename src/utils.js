import axios from "axios";

async function getRandomQuote() {
	const resp = await axios.get(
		"https://inspo-quotes-api.herokuapp.com/quotes/random"
	);

	return resp.data.quote;
}

export { getRandomQuote };
