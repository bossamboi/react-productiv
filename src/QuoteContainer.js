import { React, useState } from "react";
import Quote from "./Quote";
import { getRandomQuote } from "./utils";

/** Quote generator logic
 *
 * state: quote
 *
 * props:
 */

function QuoteContainer() {
	const [quote, setQuote] = useState(null);
	const [btnText, setBtnText] = useState(
		"Click here for an inspirational qu√∏te!"
	);

	async function handleClick() {
		const newQuote = await getRandomQuote();

		setQuote(newQuote);
		console.log("***click**");
		setBtnText("Nu quote!");
	}

	return (
		<div className="container text-end pb-3">
			<Quote quote={quote} />
			<button
				onClick={handleClick}
				style={{ color: "white", backgroundColor: "brown" }}
			>
				{btnText}
			</button>
		</div>
	);
}

export default QuoteContainer;
