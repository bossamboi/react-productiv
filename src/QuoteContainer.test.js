import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import QuoteContainer from "./QuoteContainer";
import * as utils from "./utils";

utils.getRandomQuote = jest.fn(); //mock random quote

describe("Quote container", function () {
	it("renders without crashing", function () {
		render(<QuoteContainer />);
	});

	it("contains expected default button text", function () {
		const result = render(<QuoteContainer />);
		expect(
			result.queryByText("Click here for an inspirational quøte!")
		).toBeInTheDocument();
	});

	it("does not render a quote at first render", function () {
		const { container } = render(<QuoteContainer />);
		expect(container.querySelector(".Quote-quote")).not.toBeInTheDocument();
	});

	it("renders quote and changes button text on first button click", async function () {
		await utils.getRandomQuote.mockReturnValueOnce({
			text: "Test Quote",
			author: "Test Author",
		});

		const result = render(<QuoteContainer />);
		const container = result.container;

		const quoteBtn = container.querySelector("button");
		// act(() => {
		//     fireEvent.click(quoteBtn);
		// });
		fireEvent.click(quoteBtn);

		// console.log(result);

		// test button text changed
		// expect(result.queryByText("Test Quote -Text Author")).toBeInTheDocument();
		// expect(result.queryByText("Nu quote!")).toBeInTheDocument();
		expect(container.querySelector("button")).toContain("Nu quote!");
		// expect(
		// 	result.queryByText("Click here for an inspirational quøte!")
		// ).toBeInTheDocument();
	});

	it("matches snapshot", function () {
		const { container } = render(<QuoteContainer />);
		expect(container).toMatchSnapshot();
	});
});
