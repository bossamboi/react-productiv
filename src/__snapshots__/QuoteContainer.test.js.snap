// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote container matches snapshot 1`] = `
<div>
  <div
    class="container text-end pb-3"
  >
    <div />
    <button
      style="color: white; background-color: brown;"
    >
      Click here for an inspirational qu√∏te!
    </button>
  </div>
</div>
`;
